x/pylons/keeper/msg_server_execute_recipe.go

```diff
@@ -158,6 +158,14 @@ func (k msgServer) ExecuteRecipe(goCtx context.Context, msg *types.MsgExecuteRec
 
 	id := k.AppendPendingExecution(ctx, execution, recipe.BlockInterval)
 
+	if len(msg.PaymentInfos) != 0 {
```

Comment:
Thank you!
The code in that section binds payments with pending execution to avoid double-spending. So if tx err, no pending execution so user can reuse payment for later execution.
Or we can add payments attached to refunds right at the recipe execution step
How do you think about what?

Reply:
Isn't the payment associated with the recipe_id already because of how it was created on the Stripe end? It's a cool idea to re-use it, but I think it's probably pretty unlikely to happen, it would have to be for the same item, right?